Remove support for ROCm < 5.6.0

atamazov opened the pull request:
After recent changes to MIOpen, it cannot be built with ROCm < 5.7.0. Let's remove the useless code.

----
[Attribution] @junliume @JehandadKhan
- https://github.com/ROCm/MIOpen/labels/urgency_normal
- Proposed reviewers: @junliume @JehandadKhan

atamazov committed:
Remove support for ROCm < 5.7.0

V6ser commented on src/hip/handlehip.cpp:
```diff
@@ -57,24 +56,17 @@
-#define MIOPEN_WORKAROUND_ROCM_COMPILER_SUPPORT_ISSUE_30 \
-    (MIOPEN_USE_COMGR && BUILD_SHARED_LIBS && (HIP_PACKAGE_VERSION_FLAT < 4003000000ULL))
-
+/// GFX906 and GFX103X are deprecated since 5.7 RC.
```
Why exactly was GFX103X deprecated?

atamazov replied:
@V6ser Not in MIOpen, but in HIP runtime. I have no idea why. It still works, but I see some quirks like this one.

junliume replied:
@atamazov I would recommend remove or revise this line. It's a bug and issue on hip runtime nevertheless.

atamazov replied:
@junliume I agree. Please expect update soon.

junliume commented:
@atamazov it seems that FIN updates are needed for this PR too:
```
[2024-01-13T07:51:06.284Z] /var/jenkins/workspace/MLLIBS_MIOpen_PR-2665/src/ocl/gcn_asm_utils.cpp:181:5: error: 'ROCM_FEATURE_ASM_REQUIRES_NO_XNACK_OPTION' is not defined, evaluates to 0 [-Werror,-Wundef]

[2024-01-13T07:51:06.284Z] #if ROCM_FEATURE_ASM_REQUIRES_NO_XNACK_OPTION

[2024-01-13T07:51:06.284Z]     ^
```

atamazov commented:
@junliume @cderb 
> @atamazov it seems that FIN updates are needed for this PR too:

sudden fin
![image](https://github.com/ROCm/MIOpen/assets/12697412/8bf6f26a-7aed-41be-ba02-4b420bc4b946)

atamazov committed:
deprecate-rocm-less-5.7(02) Merge branch 'develop' into deprecate-rocm-less-5.7

# RESOLVED Conflicts:
#	src/CMakeLists.txt

atamazov committed:
deprecate-rocm-less-5.7(03) Leftover that fixes build error with "-Werror"

atamazov committed:
deprecate-rocm-less-5.7(04) Resolve review comment

atamazov commented:
@junliume BTW this issue was not related to fin. Fixed! Ready for CI testing and review.

junliume commented:
@atamazov could you resolve the above conflicts? Thanks!

atamazov committed:
deprecate-rocm-less-5.7(05) Merge branch 'develop' into deprecate-rocm-less-5.7

# RESOLVED Conflicts:
#	src/hip/hip_build_utils.cpp
#	src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_padded_gemm.cpp
#	test/conv_common.hpp

atamazov commented:
@junliume Done! Ready for CI testing and review.
